packages/logging-winston/src/index.js

```diff
@@ -16,174 +16,153 @@
- *     true)
- * @property {number=} maxRetries - Maximum number of automatic retries
- *     attempted before returning the error. (default: 3)
- * @property {function=} promise - Custom promise module to use instead
- *     of native Promises.
- */
+  options = options || {};
 
-/**
- * Constructs a new object which can be used as a Winston transport.
- * @param {OptionsObject=} options - optional configuration object.
- * @constructor
- */
-var StackdriverTransport =
-    winston.transports.StackdriverLogging = function(options) {
-      options = options || {};
-      var logName = options.logName || 'winston_log';
+  var logName = options.logName || 'winston_log';
 
-      var transportOptions = {level: options.level, name: logName};
-      Transport.call(this, transportOptions);
+  winston.Transport.call(this, {
+    level: options.level,
+    name: logName
+  });
 
-      this.levels_ = options.levels || npmLevels;
-      this.metadata_ = {logName: logName};
-      if (options.resource) {
-        this.metadata_.resource = _.assign({}, options.resource);
-      }
-      var logging = require('@google-cloud/logging')(options);
-      this.logger_ = logging.log(logName);
-    };
+  this.levels_ = options.levels || NPM_LEVEL_NAME_TO_CODE;
+  this.log_ = logging(options).log(logName);
+  this.resource_ = options.resource;
+}
 
-util.inherits(StackdriverTransport, Transport);
+util.inherits(LoggingWinston, winston.Transport);
 
 /**
- * Relays a log entry to the logging agent. This is normally called by winston.
- * @param {string} level - The severity level at which this entry is being
+ * Relay a log entry to the logging agent. This is normally called by winston.
+ *
+ * @param {string} levelName - The severity level at which this entry is being
  *     logged. This should match one of the levels provided to the constructor
  *     which defaults to npm levels. This level will be translated to the
  *     appropriate Stackdriver logging severity level.
  * @param {string} msg - The message to be logged.
- * @param {object=} meta - Winston provided metadata that should be attached to
- *     log entry. Each property will be converted to a string using
- *     `util.inspect`.
- * @param {function=} callback A callback that is invoked when the logging agent
- *     either succeeds or gives up writing the log entry to the remote server.
+ * @param {object=} metadata - Winston-provided metadata that should be attached
+ *     to the log entry. Each property will be converted to a string using
+ *     `JSON.stringify`.
+ * @param {function=} callback - A callback that is invoked when the logging
+ *     agent either succeeds or gives up writing the log entry to the remote
+ *     server.
  */
-StackdriverTransport.prototype.log = function(level, msg, meta, callback) {
-  if (_.isFunction(meta) && !callback) {
-    callback = meta;
-    meta = null;
+LoggingWinston.prototype.log = function(levelName, msg, metadata, callback) {
```

Comment:
That was confusing indeed. I went the other way around and instead of changing `metadata` to an object expected by `log.entry()`, I created a new var named `entryMetadata`. PTAL.